### PR TITLE
fix(vscode): server bin path resolution

### PR DESCRIPTION
**Pull request by nhedger**
## Summary

This PR fixes an issue where the VS Code extension cannot resolve the path to the biome CLI when using certain package managers, notably pnpm.

The new strategy is to point to the `node_modules/.bin/biome` path, which is consistent for all package managers.

## Test Plan

- [x] Tested working with npm on:
	- [x] macOS
	- [x] Windows
	- [x] Linux
- [x] Tested working with pnpm on:
	- [x] macOS
	- [x] Windows
	- [x] Linux
- [x] Tested working with yarn on:
	- [x] macOS
	- [x] Windows
	- [x] Linux
- [x] Tested working with bun on:
	- [x] macOS
	- [x] ~Windows~ (not applicable at the moment)
	- [x] Linux

---

I'd like some help to test under Windows, I don't have a machine handy.

### CODE REVIEW

**Comment by fox1t** [comment]
Could you add an automation test on the CI to test this code properly?

**Comment by nhedger** [comment]
> Could you add an automation test on the CI to test this code properly?

It is technically possible, and while I'm usually all for it, this would require a non-trivial amount of additional work, for which I'm not willing to invest time at the moment.

Considering that the VS Code extension rarely changes, I think that doing a manual test will suffice for now. Feel free to PR something if/when this is merged.

**Comment by SuperchupuDev** [comment]
what's left for this pull request to be considered ready for review? i'd love it if it managed to get included in the next biome release

**Comment by nhedger** [comment]
It needs to be tested under Windows and Linux.

I'll come up with a detailed test plan tomorrow.

File: editors/vscode/src/main.ts [hunk 1]
```diff
@@ -250,37 +252,22 @@ async function getWorkspaceDependency(
...
-			console.log(e);
-			window.showWarningMessage(`The extension couldn't resolve ${manifestName} or ${binaryName}.
-			If you installed "@biomejs/biome", it's a resolving issue due to your package manager. Check the troubleshooting section of the extension for more information on how to fix the issue.`);
+		if (await fileExists(path)) {
+			return path.fsPath;
 		}
 	}
 
+	window.showWarningMessage(
+		"Unable to resolve the biome server from your dependencies.",
```
Comment (ematipico) [inline editors/vscode/src/main.ts:268]:
```suggestion
		"Unable to resolve the biome server from your dependencies. Make sure it's correctly installed, or untick the `requireConfiguration` setting to use the bundled binary.",
```

**Comment by nhedger** [comment]
Alright, it works as expected on Linux. I'll be testing on a Windows VM today and then we should be good to go.

**Comment by nhedger** [comment]
A little adjustment was needed to make it work with Windows, but it should be all good now. I'll mark the PR as ready and will let review the code.

File: editors/vscode/package.json [hunk 2]
```diff
@@ -135,7 +135,7 @@
 	"license": "MIT",
 	"scripts": {
 		"compile": "esbuild src/main.ts --bundle --outfile=out/main.js --external:vscode --format=cjs --platform=node --target=node14",
-		"watch": "pnpm run compile -- --sourcemap --watch",
```
Comment (nhedger) [inline editors/vscode/package.json:138]: Missed this one in https://github.com/biomejs/biome/pull/219.

Let me know if you'd rather I submit this in a separate PR.

File: editors/vscode/package.json [hunk 3]
```diff
@@ -135,7 +135,7 @@
...
+		"watch": "pnpm run compile --sourcemap --watch",
```
Comment (nhedger) [inline editors/vscode/package.json:138]: Missed this one in https://github.com/biomejs/biome/pull/219.

Let me know if you'd rather I submit this in a separate PR.

**Review by ematipico — APPROVED** [review]
